erpnext/assets/doctype/asset/asset.py

```diff
@@ -17,12 +17,12 @@ def validate(self):
 		self.validate_item()
 		self.set_missing_values()
 		self.validate_asset_values()
-		self.make_depreciation_schedule()
-		self.set_accumulated_depreciation()
+		if self.calculate_depreciation:
```

Comment:
Is this a new field?

Reply:
@nabinhait Yes, it is a checkbox. We set it in Asset if Depreciation should be calculated or not. Some assets does not depreciate like Leased assets or Land or softwares.
<img width="920" alt="screen shot 2017-11-30 at 4 54 00 pm" src="https://user-images.githubusercontent.com/16913064/33428502-1e96420a-d5ef-11e7-840b-5faa3ea606e5.png">